repos/spack_repo/builtin/packages/charliecloud/package.py

```diff
@@ -11,15 +11,17 @@ class Charliecloud(AutotoolsPackage):
     """Lightweight user-defined software stacks for HPC."""
 
     maintainers("j-ogas", "reidpr", "loshak")
-    homepage = "https://hpc.github.io/charliecloud"
-    url = "https://github.com/hpc/charliecloud/releases/download/v0.18/charliecloud-0.18.tar.gz"
-    git = "https://github.com/hpc/charliecloud.git"
+    homepage = "https://charliecloud.io/"
+    url = "https://gitlab.com/charliecloud/main/-/archive/v0.42/main-v0.42.tar.gz"
```

Comment:
```suggestion
    url = "https://gitlab.com/charliecloud/charliecloud/-/archive/v0.42/main-v0.42.tar.gz"
```

because of a recent rename ([#1999](https://gitlab.com/charliecloud/charliecloud/-/issues/1999)).